examples/Differential evolution.ipynb

```diff
@@ -0,0 +1,276 @@
+      "text/plain": [
+       "<matplotlib.figure.Figure at 0x1e6ab574ef0>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "import sys\n",
+    "import os\n",
+    "cwd = os.getcwd()\n",
+    "sys.path.append('C://Users//bclamber//Documents//GitHub//pints')\n",
+    "from __future__ import print_function\n",
+    "import pints\n",
+    "import pints.toy as toy\n",
+    "import pints._diagnostics as diagnostics\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "# Load a forward model\n",
+    "model = toy.LogisticModel()\n",
+    "\n",
+    "# Create some toy data\n",
+    "real_parameters = [0.015, 500]\n",
+    "times = np.linspace(0, 1000, 1000)\n",
+    "org_values = model.simulate(real_parameters, times)\n",
+    "\n",
+    "# Add noise\n",
+    "noise = 10\n",
+    "values = org_values + np.random.normal(0, noise, org_values.shape)\n",
+    "real_parameters = np.array(real_parameters + [noise])\n",
+    "\n",
+    "# Get properties of the noise sample\n",
+    "noise_sample_mean = np.mean(values - org_values)\n",
+    "noise_sample_std = np.std(values - org_values)\n",
+    "\n",
+    "# Create an object with links to the model and time series\n",
+    "problem = pints.SingleSeriesProblem(model, times, values)\n",
+    "\n",
+    "# Create a log-likelihood function (adds an extra parameter!)\n",
+    "log_likelihood = pints.UnknownNoiseLogLikelihood(problem)\n",
+    "\n",
+    "# Create a uniform prior over both the parameters and the new noise variable\n",
+    "prior = pints.UniformPrior(\n",
+    "    [0.01, 400, noise*0.1],\n",
+    "    [0.02, 600, noise*100]\n",
+    "    )\n",
+    "\n",
+    "# Create a Bayesian log-posterior (prior * likelihood)\n",
+    "log_posterior = pints.LogPosterior(prior, log_likelihood)\n",
+    "\n",
+    "# Create differential evolution MCMC routine\n",
+    "x0 = real_parameters * 1.1\n",
+    "mcmc = pints.DreamMCMC(log_posterior, x0)\n",
+    "\n",
+    "# Use 4000 iterations in total\n",
+    "mcmc.set_iterations(4000)\n",
+    "\n",
+    "# Discard the first 2000 iterations as burn in\n",
+    "mcmc.set_burn_in(2000)\n",
+    "\n",
+    "# Store only every 4th sample\n",
+    "mcmc.set_thinning_rate(4)\n",
+    "\n",
+    "# Disable verbose mode\n",
+    "mcmc.set_verbose(False)\n",
+    "\n",
+    "# Set gamma\n",
+    "mcmc.set_gamma(0.1)\n",
+    "\n",
+    "# Run!\n",
+    "print('Running...')\n",
+    "chains = mcmc.run()\n",
+    "print('Done!')\n",
+    "\n",
+    "# Plot output\n",
+    "import pints.plot # Use Pints' diagnostic tool\n",
+    "\n",
+    "stacked = np.vstack(chains)\n",
+    "pints.plot.pairwise(stacked, kde=False)\n",
+    "\n",
+    "plt.show()"
+ 
```

Comment:
We probably want the examples to be more consistent. We can either split the two examples or we can also do one that use the same data and compare the performance of different MCMC algorithms?

I will split the two first.